### DISCUSSION TITLE
Improvement thoughts

### DISCUSSION THREAD

**Discussion by Kh3nsu** [Radiokot/photoprism-android-client#83] [Ideas]
Hello, 

First of all thank you for designing a gallery app for photoprism! I was looking for one for so long now. This is one of the best attempts I have used so far. Even replaced the PWA of PhotoPrism this time.

I know the gallery app is quite new and is probably und heavy development still. But I got some Improvements which may would improve the app even more:

- Caching of last Pictures (Maybe only thumbnail and with a size limit or time frame) Videos should only be cached by its thumbnail, As caching Videos could be quite storage intensive.
- separate albums and folders. This has the background as my original folders are separated with year/month/folder name of the device. It shows me every single folder and the albums combined which is quite a lot and clutters the album view unnecessarily. I don't need folders at all so I also would separate the buttons to activate it.
- View Metadata of picture by swiping up while being in the picture. It could show location, time etc. (This was already mentioned in another discussion.)

Some default features which probably are already on your roadmap but low priority as far as I understood your goals.
- Support deleting (I know in photorpism you first have to archive it and delete it afterwards (Which is really annoying), maybe this could be automated in the gallery app so quick deletion is possible)
- Support Archiving
- Support marking as favorite
- Album creation / editing / viewing, directly adding pictures to album

I'm not sure how limited the API of PhotoPrism is or how you have to interact with it. I just wanted to share some thoughts what could be done in the future. 

I would love to hear a feedback :)

Best regards,
Kevin

**Comment by Radiokot** [comment]
Hi Kevin. Thanks for the many ideas on how to improve the gallery 😊
1. Currently, all the graphical content is saved into the cache of limited size, which displaces the least recently used items. You still need connection to load the library data. I'm not going to implement the offline mode in the gallery, and there are no API capabilities to do that for now;
2. I agree that disabling loading of folders could be useful. Will consider adding it;
3. I don't plan on adding the in-app metadata viewer, at least in the nearest future. It is already possible to see the metadata by clicking the 🌍 button, which opens the item in the PWA;
4. Deleting, archiving, liking/disliking – could be useful, will consider adding these. But again, it is currently possible to do all this by clicking the 🌍 button;
5. I don't plan on adding any heavy library-management features, like management of albums. My goal is to make the gallery a useful, simple and stable viewer + to fill the lack of common features in the PhotoPrism web app.

**Comment by Radiokot** [comment]
By the way, you can make your favorite albums always be at the beginning of the list – simply star them in the web app. Learn more about the search from the guide: https://github.com/Radiokot/photoprism-android-client/wiki/How-to-search-the-library

**Comment by Radiokot** [comment]
The version [1.19.0](https://github.com/Radiokot/photoprism-android-client/releases/tag/1.19.0(29)) brings the ability to disable showing folders in the album list.

**Comment by Kh3nsu** [comment]
Thanks for the quick update! I haven't got the 1.19 on play store yet.

I just noticed one more thing:
When vieweing a long video the display turns off after some while and you have to tap it to prevent that. Normally this does not happen when watching videos.

**Reply by Radiokot** [reply]
Wow, this is a bug. The display should be on while a video is playing. Thanks for reporting.

**Reply by Radiokot** [reply]
[1.20.0](https://github.com/Radiokot/photoprism-android-client/releases/tag/1.20.0(30)) release contains the screen timeout fix.

**Comment by Kh3nsu** [comment]
Just got a two more ideas. 

-What do you think of replaying live pictures as long as you hold down your finger onto it?
Currently it is playing once after you click on it.

- Selecting many pictures could be made easier by holding and swiping to select more in one go.
 (Especially important for archiving / deleting feature in future releases)

**Reply by Radiokot** [reply]
Good ideas 👍🏻

**Reply by Radiokot** [reply]
I thought about selection by moving the finger while pressing, but I don't know what how to handle items with multiple files (video, RAW, vector, etc.) in this case.

**Reply by Kh3nsu** [reply]
Yeah that would be great. I mean I only enabled original download in photoprism without the rest. I would just say you do a cut here and say that it should download the original file without all the other stuff. Keep it as easy as possible. Or did I get that wrong?

**Reply by Radiokot** [reply]
![Screenshot_20231024_181600_PhotoPrism Gallery](https://github.com/Radiokot/photoprism-android-client/assets/5675681/ec4175c4-2d8c-4ce1-91a3-a8e0301f3818)
I mean this case, where it is up to the user what to select

**Comment by Radiokot** [comment]
Hi. The recent [1.21.0](https://github.com/Radiokot/photoprism-android-client/releases/tag/1.21.0(34)) update contains the ability to like/dislike photos as well as see only liked (favorite) content.

**Reply by Kh3nsu** [reply]
That's awesome! Is it Hard to implement bulk marking to like/dislike photos? As it is possible with downloading and sharing already?

**Reply by Radiokot** [reply]
Do you think it would be useful?

**Reply by Kh3nsu** [reply]
Well it won't hurt I guess, I sometimes go through my new pictures and select the ones looking the best. But I also open them so I could directly mark them as favorite. I don't know. I think bulk deleting / archiving would be more useful.

**Comment by Radiokot** [comment]
Batch deleting and archiving is available in [1.27.0](https://github.com/Radiokot/photoprism-android-client/releases/tag/1.27.0(41))
